presto-main/src/main/java/com/facebook/presto/server/testing/TestingPrestoServer.java

```diff
@@ -311,13 +331,24 @@ public TestingPrestoServer(
         if (coordinator) {
             dispatchManager = injector.getInstance(DispatchManager.class);
             queryManager = (SqlQueryManager) injector.getInstance(QueryManager.class);
-            resourceGroupManager = Optional.of(injector.getInstance(InternalResourceGroupManager.class));
+            ResourceGroupManager<?> rm = injector.getInstance(ResourceGroupManager.class);
+            resourceGroupManager = rm instanceof InternalResourceGroupManager ? Optional.of((InternalResourceGroupManager<?>) rm) : Optional.empty();
             nodePartitioningManager = injector.getInstance(NodePartitioningManager.class);
             planOptimizerManager = injector.getInstance(ConnectorPlanOptimizerManager.class);
             clusterMemoryManager = injector.getInstance(ClusterMemoryManager.class);
             statsCalculator = injector.getInstance(StatsCalculator.class);
             eventListenerManager = ((TestingEventListenerManager) injector.getInstance(EventListenerManager.class));
         }
+        else if (resourceManager) {
+            dispatchManager = null;
```

Comment:
wondering if we need to explicitly set this to null.

Reply:
This is a final member variable so it needs to be explicitly set.